### DISCUSSION TITLE
cannot get cron working on clients

### DISCUSSION THREAD

**Discussion by schroenki** [ltsp/ltsp#454] [Q&A] [answered]
Hello everyone, I am running an LTSP setup using the server as template for the image (ltsp image /). I tried to get the clients to shutdown at 17:30 every day by adding the following line to ltsp.conf, as described in the documentation:
`CRONTAB_x="30 17 * * * /usr/sbin/shutdown now"`
This is also reflected on the clients, as the output of
`cat /etc/cron.d/ltsp`
is
`# Generated by 'ltsp init', see man:ltsp(8)
30 17 * * * /usr/sbin/shutdown now`
However the clients do nothing at 17:30. Cron is installed and running on the server, cronjobs run properly.
I can't figure out why it is not working, I do not have many customisation on the image and followed the documentation closely.
Any help would be appreciated!

**Comment by alkisg** [comment, marked as answer]
Hi, try to add the username for the cronjob:

```shell
 CRONTAB_x="30 17 * * * root poweroff"
```

**Reply by schroenki** [reply]
Thanks for the reply! Will try that now. (As soon as I know if the clients powered down I will let you know if it worked.)

**Reply by schroenki** [reply]
As said below, that was the solution, thank you, marked as answer.

**Comment by fottsia** [comment]
I confirm that adding "root" made CRONTAB work for me.
It would be nice to also fix the ltsp.conf man page example.

**Comment by alkisg** [comment]
Note that by default, PATH was limited to /usr/bin:/bin, so plain "poweroff" wasn't working, and "/usr/sbin/poweroff" was needed.

Fixed both the PATH and the man page example in commit e55802b.

**Comment by schroenki** [comment]
Can also confirm that adding root made it work - thank you for all your answers, works perfectly now! I also wanted to suggest adapting the manpage example accordingly, but that was already suggested by fottsia.
However, I don't quite get why adding root is necessary - I am pretty sure I checked whether a non-root user can run shutdown and it worked. (I will check that again tomorrow)

**Comment by alkisg** [comment]
I.e. the problem isn't related to permissions, it's related to the cron file formats. There are two of them.

The cron files at /etc/cron.d, that ltsp uses, need a user name column.
While `crontab -e` saves its files at /var/spool/cron/crontabs. Since these files ARE per-user, the use a different file format; they don't have a user column.
